sdks/python/apache_beam/transforms/combiners.py

```diff
@@ -159,82 +159,112 @@ class Top(object):
+        return (
+            pcoll
+            | core.ParDo(_TopPerBundle(self._n, compare))
+            | core.GroupByKey()
+            | core.ParDo(_MergeTopPerBundle(self._n, compare)))
+      else:
+        return pcoll | core.CombineGlobally(
+            TopCombineFn(self._n, compare, self._key, self._reverse),
+            *self._args, **self._kwargs)
 
-  @staticmethod
-  @ptransform.ptransform_fn
-  def PerKey(pcoll, n, compare=None, *args, **kwargs):
+  class PerKey(ptransform.PTransform):
     """Identifies the compare-most N elements associated with each key.
 
     This transform will produce a PCollection mapping unique keys in the input
     PCollection to the n greatest elements with which they are associated, where
     "greatest" is determined by the comparator function supplied as the compare
-    argument.
-
-    compare should be an implementation of "a < b" taking at least two arguments
-    (a and b). Additional arguments and side inputs specified in the apply call
-    become additional arguments to the comparator.  Defaults to the natural
-    ordering of the elements.
-
-    The arguments 'key' and 'reverse' may instead be passed as keyword
-    arguments, and have the same meaning as for Python's sort functions.
-
-    Args:
-      pcoll: PCollection to process.
-      n: number of elements to extract from pcoll.
-      compare: as described above.
-      *args: as described above.
-      **kwargs: as described above.
-
-    Raises:
-      TypeCheckError: If the output type of the input PCollection is not
-        compatible with KV[A, B].
+    argument in the initializer.
     """
-    key = kwargs.pop('key', None)
-    reverse = kwargs.pop('reverse', False)
-    return pcoll | core.CombinePerKey(
-        TopCombineFn(n, compare, key, reverse), *args, **kwargs)
+    def __init__(self, n, compare=None, *args, **kwargs):
+      """Initializer.
+
+      compare should be an implementation of "a < b" taking at least two
+      arguments (a and b). Additional arguments and side inputs specified in
+      the apply call become additional arguments to the comparator.  Defaults to
+      the natural ordering of the elements.
+
+      The arguments 'key' and 'reverse' may instead be passed as keyword
+      arguments, and have the same meaning as for Python's sort functions.
+
+      Args:
+        n: number of elements to extract from input.
+        compare: as described above.
+        *args: as described above.
+        **kwargs: as described above.
+      """
+      self._n = n
+      self._compare = compare
+      self._key = kwargs.pop('key', None)
+      self._reverse = kwargs.pop('reverse', False)
+      self._args = args
+      self._kwargs = kwargs
+
+    def default_label(self):
+      return 'TopPerKey(%d)' % self._n
+
+    def expand(self, pcoll):
+      """Expands the transform.
+
+      Raises TypeCheckError: If the output type of the input PCollection is not
```

Comment:
do you need to break it into two lines?

Raises:
  TypeCheckError: ...

Reply:
No, intentionally moved this to a single line. 

doc checker was failing because of a missing reference to `TypeCheckError`. It is likely a configuration error but did not want to block the change on that.